yt_dlp/extractor/servus.py

```diff
@@ -1,148 +1,304 @@
 # coding: utf-8
 from __future__ import unicode_literals
 
+import json
+from operator import itemgetter
+
 from .common import InfoExtractor
+from ..compat import (
+    compat_parse_qs,
+    compat_urllib_parse_unquote_plus,
+    compat_urllib_parse_urlparse,
+)
 from ..utils import (
-    determine_ext,
-    float_or_none,
-    int_or_none,
-    unified_timestamp,
-    urlencode_postdata,
-    url_or_none,
+    get_element_by_id,
+    parse_iso8601,
+    traverse_obj,
+    ExtractorError,
+    GeoRestrictedError,
+    OnDemandPagedList,
 )
 
 
-class ServusIE(InfoExtractor):
+class ServusTVIE(InfoExtractor):
+    IE_NAME = 'servustv'
     _VALID_URL = r'''(?x)
                     https?://
-                        (?:www\.)?
-                        (?:
-                            servus\.com/(?:(?:at|de)/p/[^/]+|tv/videos)|
-                            (?:servustv|pm-wissen)\.com/videos
-                        )
-                        /(?P<id>[aA]{2}-\w+|\d+-\d+)
+                        (?:www\.)?servustv.com
+                        /[\w-]+/(?:v|[bp]/[\w-]+)
+                        /(?P<id>[A-Za-z0-9-]+)
                     '''
+    _GEO_COUNTRIES = ['AT', 'DE', 'CH', 'LI', 'LU', 'IT']
+    _GEO_BYPASS = False
+    _API_URL = 'https://api-player.redbull.com/stv/servus-tv'
+    _QUERY_API_URL = 'https://backend.servustv.com/wp-json/rbmh/v2/query-filters/query/'
+    _LIVE_URLS = {
+        'AT': 'https://dms.redbull.tv/v4/destination/stv/stv-linear'
+              '/personal_computer/chrome/at/de_AT/playlist.m3u8',
+        'DE': 'https://dms.redbull.tv/v4/destination/stv/stv-linear'
+              '/personal_computer/chrome/de/de_DE/playlist.m3u8',
```

Comment:
read contributing guidelines. It specifically says DO NOT split literals like this
```suggestion
        'AT': 'https://dms.redbull.tv/v4/destination/stv/stv-linear/personal_computer/chrome/at/de_AT/playlist.m3u8',
        'DE': 'https://dms.redbull.tv/v4/destination/stv/stv-linear/personal_computer/chrome/de/de_DE/playlist.m3u8',
```

also, it appears only the language code in the URL is different. In that case, a mapping like this is unnecessary